lib/shoulda/matchers/active_record/have_db_index_matcher.rb

```diff
@@ -124,7 +125,9 @@ def correct_unique?
         end
 
         def matched_index
-          indexes.detect { |each| each.columns == @columns }
+          indexes.detect do |each|
+            each.columns == @columns || normalize_columns_to_string(each.columns) == @joined_columns
```

Comment:
Can you break this line after the `||`?

Also, what do you think about normalizing the `indexes` array and always using `@joined_columns` rather than `@columns` rather than calling `normalize_columns_to_string` on each iteration? Would the tests still pass?

Reply:
Ha - sorry - can you break this line after the `||`?

Alternatively, what do you think about normalizing `indexes` (and always using `@joined_columns` rather than `@columns`) instead of calling `normalize_columns_to_string` in each iteration. Would the tests still pass?

Reply:
The issue with always comparing the join version, is then the spec defined becomes order depended. Currently, as an array, it is ordered independent for most default cases. The specs would still pass, I can add a spec for the order-independence.